testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_transactions.py

```diff
@@ -0,0 +1,420 @@
+@pytest.mark.cbl
+def test_transactions_with_latest_updates(params_from_base_test_setup):
+    '''
+    @summary:
+    https://docs.google.com/spreadsheets/d/1YWP_iwYz9B3ej_ksLf-XKsLj4RnbuSigf7hp0BPYv0w/edit#gid=0
+    Covered #13
+    1. Inserts a series of documents (A, B, C) in a transaction
+    2. CBL is not syncing as yet. SGW does import
+    3. Transaction updates (edits) a series of documents in this order  (A->A',B-> B',C-> C')  At the exact same time, CBL starts pulling docs via SGW.
+    4. Replicate from SGW to CBL
+    5. Verify CBL eventually pull all updated docs
+    '''
+
+    xattrs_enabled = params_from_base_test_setup['xattrs_enabled']
+    if not xattrs_enabled:
+        pytest.skip('this test require --xattrs flag')
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    db = params_from_base_test_setup["db"]
+    cluster_helper = ClusterKeywords(cluster_config)
+    topology = cluster_helper.get_cluster_topology(cluster_config)
+    cluster_servers = topology["couchbase_servers"]
+    cbs_one_url = cluster_servers[0]
+    cb_server = couchbaseserver.CouchbaseServer(cbs_one_url)
+
+    sg_db = "db"
+    username = "autotest"
+    password = "password"
+    channels = ["TRANSACTIONS"]
+    num_of_docs = 4
+    doc_ids = ""
+
+    sg_client = MobileRestClient()
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+    bucket = cb_server.get_bucket_names()[0]
+
+    # disable replicas and rebalance the server
+    cb_server.disable_replicas(bucket)
+    cb_server.rebalance_server(cluster_servers)
+
+    # randomly  generate replication ids
+    for i in range(0, num_of_docs):
+        doc_id = "txn_id_" + str(time.time())
+        doc_ids = doc_id + "," + doc_ids
+    opt_doc_ids = doc_ids.replace("txn_id_", "insert-txn_id_")
+    # 1. Inserts a series of documents (A, B, C) in a transaction
+    # 2. CBL is not syncing as yet. SGW does import
+    transactions_app_dir = "resources/data/transaction-performer.jar"
+    doc_body = document.create_doc(doc_id=doc_id, content="doc1", channels=channels)
+    data = json.dumps(doc_body)
+    cmd = "java -cp {} com.couchbase.transaction clusterip={} operationdocids={} bucket={} doccontent='{}'".format(transactions_app_dir, c.servers[0].host, opt_doc_ids, bucket, data)
+    subprocess.check_output(cmd, shell=True)
+
+    # 3. Transaction updates (edits) a series of documents in this order  (A->A',B-> B',C-> C')  At the exact same time, CBL starts pulling docs via SGW.
+    # replace_doc_body = document.create_doc(doc_id=doc_id, content="updated_doc", channels=channels, cbl=True)
```

Comment:
clean up the commented code if not needed.

Reply:
done